design/node-agent-concurrency.md

```diff
@@ -0,0 +1,78 @@
+- The resources may be different among nodes. Then users could specify smaller concurrent number for nodes with less resources while larger number for the ones with more resources
+- Help users to isolate critical environments. Users may run some critical workloads in some specified nodes, since VGDP instances may take large resource consumption, users may want to run less number of instances in the nodes with critical workloads
+
+The range of Per-node concurrent number is the same with Global concurrent number.  
+Per-node concurrent number is preferable to Global concurrent number, so it will overwrite the Global concurrent number for that node.  
+Per-node concurrent number is implemented in a separate configMap named ```node-agent-configs```. This configMap is not created by Velero, users should create it manually on demand. The configMap should be created before node-agent server starts, otherwise, node-agent server should be restarted in order to make the effect.  
+
+The ```node-agent-configs``` configMap may be used for other purpose of configuring node-agent in future, at present, there is only one kind of data ```data-path-concurrency```. Its data struct is like below: 
+
+```go
+type DataPathConcurrency struct {
+	// ConfigRules specifies the configs by node identified by label selector
+	ConfigRules map[string]int `json:"configRules"`
+}
+```
+
+The key of map of ```ConfigRules``` is a label selector, this means, the nodes are identified by labels.  
+For example, the map could have below elements:
+```
+"kubernetes.io/hostname=node1": 3
+"beta.kubernetes.io/instance-type=Standard_B4ms": 5
+```
+The first element means the node with host name ```node1``` get the Per-node concurrent number of 3.  
+The second element means all the nodes with label ```beta.kubernetes.io/instance-type``` and value ```Standard_B4ms``` get the Per-node concurrent number of 5. 
+At least one node is expected to have a label with the specified key. If no node is with this label, the Per-node configuration makes no effect.  
+If one node falls into more than one elements, e.g., if ```node1``` also has the label ```beta.kubernetes.io/instance-type=Standard_B4ms```, the smallest number (3) will be used.  
+
+### Global data path manager
+As for the code implementation, data path manager is to maintain the total number of the running VGDP instances and ensure the limit is not excceeded. At present, there is one data path manager instance per controller, as a result, the concurrent numbers are calculated separately for each controller. This doesn't help to limit the concurrency among different requesters.  
+Therefore, we need to create one global data path manager instance server-wide, and pass it to different controllers. The instance will be created at node-agent server startup.  
```

Comment:
by server-wide, do you mean node-agent server or the whole velero setup?  I assume it's node-agent server, if that's the node-agent server, I don't think it's global

Reply:
It is node-agent server. Maybe we can give it a better name, what about "Ordinary Number"?